tests/app_metadata/main.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the low-level I2C peripheral driver
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Kevin Weiss <kevin.weiss@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "shell.h"
+#include "test_helpers.h"
+
+#ifndef PARSER_DEV_NUM
+#define PARSER_DEV_NUM 0
+#endif
+
+void print_app_metadata(int dev)
+{
+    print_cmd(dev,"app_metadata()");
+    print_data_dict_str(dev, "app_name", RIOT_APPLICATION);
+    print_data_dict_str(dev, "board", RIOT_BOARD);
+    print_data_dict_str(dev, "cpu", RIOT_CPU);
+#ifdef APP_SHELL_FMT
+    print_data_dict_str(dev, "app_shell_fmt", APP_SHELL_FMT);
+#endif
+    print_data_dict_str(dev, "mcu", RIOT_MCU);
+    print_data_dict_str(dev, "os_version", RIOT_VERSION);
+    print_result(dev, TEST_RESULT_SUCCESS);
+}
+
+int cmd_app_metadata(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    print_app_metadata(PARSER_DEV_NUM);
+    return 0;
+}
+
+int cmd_test_result_only(int argc, char **argv)
+{
+    (void)argv;
+    if (argc > 1) {
+        print_result(PARSER_DEV_NUM, TEST_RESULT_ERROR);
+    }
+    else {
+        print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    }
+    return 0;
+}
+
+int cmd_test_cmd(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    print_cmd(PARSER_DEV_NUM,"test_cmd()");
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    return 0;
+}
+
+int cmd_test_data_int(int argc, char **argv)
+{
+    for (int i = 1; i < argc; i++) {
+        print_data_int(PARSER_DEV_NUM, atoi(argv[i]));
+    }
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    return 0;
+}
+
+int cmd_test_data_str(int argc, char **argv)
+{
+    for (int i = 1; i < argc; i++) {
+        print_data_str(PARSER_DEV_NUM, argv[i]);
+    }
+    print_result(PARSER_DEV_NUM, TEST_RESULT_SUCCESS);
+    return 0;
+}
+
+int cmd_help(int argc, char **argv);
```

Comment:
why simply move the whole function (with ifdef) here, instead of the prototype?

Reply:
Because I need to know the size of `shell_commands[]` :stuck_out_tongue_winking_eye:

Reply:
I will iterate through until NULL